### ISSUE TITLE
@time throws "TypeError: Illegal invocation"

### ISSUE THREAD

**Issue by felskov** [jayphelps/core-decorators.js#54]
Using the example from https://www.npmjs.com/package/core-decorators#time I get "TypeError: Illegal invocation" thrown in the console:

```
class Bird {
  @time('sing')
  sing() {
  }
}
```

I looked into the sourcecode, and believe that the issue is related to the following line:

```
const CONSOLE = konsole || CONSOLE_NATIVE;
```

It appears that konsole is always null (at least in this simple usecase), thus CONSOLE is always set to the native console. Later on you do the following:

```
const time = CONSOLE.time || CONSOLE_TIME;
const timeEnd = CONSOLE.timeEnd || CONSOLE_TIMEEND;
```

And because CONSOLE is pointing to the native console, the correctly bound CONSOLE_TIME and CONSOLE_TIMEEND are not used. If I remove `CONSOLE.time(End) ||`, no errors are thrown and times are outputted to the console.

**Comment by jayphelps** [comment]
What browser is this happening in?

**Comment by felskov** [comment]
I ran into the issue using Chrome v49.0.2623.8, running on Mac OS X 10.11.1. I'm using core-decorators 0.11.0.

**Comment by jayphelps** [comment]
@felskov found the issue and fixed/published in `v0.11.1`. :+1: sorry for the trouble and thanks for reporting!

**Comment by felskov** [comment]
No worries - thanks for the fast reply
